crates/currency/src/lib.rs

```diff
@@ -14,23 +15,419 @@ use frame_support::{
+                .try_into()
+                .map_err(|_| Error::<T>::TryIntoIntError)?;
+
+            Ok(Self {
+                amount,
+                currency_id: self.currency_id,
+            })
+        }
+
+        pub fn checked_div(&self, scalar: &UnsignedFixedPoint<T>) -> Result<Self, DispatchError> {
+            let amount = UnsignedFixedPoint::<T>::checked_from_integer(self.amount)
+                .ok_or(Error::<T>::TryIntoIntError)?
+                .checked_div(&scalar)
+                .ok_or(Error::<T>::ArithmeticOverflow)?
+                .into_inner()
+                .checked_div(&UnsignedFixedPoint::<T>::accuracy())
+                .ok_or(Error::<T>::ArithmeticUnderflow)?;
+            Ok(Self {
+                amount,
+                currency_id: self.currency_id,
+            })
+        }
+
+        pub fn ratio(&self, other: &Self) -> Result<UnsignedFixedPoint<T>, DispatchError> {
+            ensure!(self.currency_id == other.currency_id, Error::<T>::InvalidCurrency);
+            let ratio = UnsignedFixedPoint::<T>::checked_from_rational(self.amount, other.amount)
+                .ok_or(Error::<T>::TryIntoIntError)?;
+            Ok(ratio)
+        }
+
+        pub fn to_fixed(&self) -> Result<SignedFixedPoint<T>, DispatchError> {
+            let signed_inner =
+                TryInto::<SignedInner<T>>::try_into(self.amount).map_err(|_| Error::<T>::TryIntoIntError)?;
+            let signed_fixed_point = <T as pallet::Config>::SignedFixedPoint::checked_from_integer(signed_inner)
+                .ok_or(Error::<T>::TryIntoIntError)?;
+            Ok(signed_fixed_point)
+        }
+
+        pub fn min(&self, other: &Self) -> Result<Self, DispatchError> {
+            ensure!(self.currency_id == other.currency_id, Error::<T>::InvalidCurrency);
+            Ok(if self.le(other)? { self.clone() } else { other.clone() })
+        }
+
+        pub fn lt(&self, other: &Self) -> Result<bool, DispatchError> {
+            ensure!(self.currency_id == other.currency_id, Error::<T>::InvalidCurrency);
+            Ok(self.amount < other.amount)
+        }
+
+        pub fn le(&self, other: &Self) -> Result<bool, DispatchError> {
+            ensure!(self.currency_id == other.currency_id, Error::<T>::InvalidCurrency);
+            Ok(self.amount <= other.amount)
+        }
+
+        pub fn eq(&self, other: &Self) -> Result<bool, DispatchError> {
+            ensure!(self.currency_id == other.currency_id, Error::<T>::InvalidCurrency);
+            Ok(self.amount == other.amount)
+        }
+
+        pub fn ge(&self, other: &Self) -> Result<bool, DispatchError> {
+            ensure!(self.currency_id == other.currency_id, Error::<T>::InvalidCurrency);
+            Ok(self.amount >= other.amount)
+        }
+
+        pub fn gt(&self, other: &Self) -> Result<bool, DispatchError> {
+            ensure!(self.currency_id == other.currency_id, Error::<T>::InvalidCurrency);
+            Ok(self.amount > other.amount)
+        }
```

Comment:
Maybe we should add a `CheckedPartialOrd` trait for these?

Reply:
Why? Imo that only makes sense if we implement this trait also for other types